metaflow/metaflow_config.py

```diff
@@ -198,4 +199,56 @@ def get_authenticated_boto3_client(module, params={}):
             except requests.exceptions.HTTPError as e:
                 raise MetaflowException(repr(e))
         return boto3.session.Session(**cached_aws_sandbox_creds).client(module, **params)
+
+    if METAFLOW_AWS_ARN:
+        import logging
+        from datetime import datetime
+
+        import boto3
+        from botocore.credentials import AssumeRoleCredentialFetcher, DeferredRefreshableCredentials
+        from botocore.session import Session
+        from dateutil.tz import tzlocal
+
+        _logger = logging.getLogger(__name__)
+
+        def get_aws_session(role_arn: str) -> Session:
+            """
+            Args:
+                role_arn: AWS ARN
+            Returns: boto3 Session
+            """
+            source_session = boto3.Session()
+
+            # Use profile to fetch assume role credentials
+            fetcher = AssumeRoleCredentialFetcher(
+                client_creator=source_session._session.create_client,
```

Comment:
Can we make sure we still address the open comments from #4 or at least mirror them here as I think the majority are still valid.

Reply:
I replied on the closed PR as well. Replying here as this is the relevant PR. 

I've modified the code to make it more readable.

I did look into whether there was a better way (as opposed to assigning to/using private variables) and it doesn't look like there's good support for this yet. I've created a task to track this in [AIP-1608](https://zbrt.atl.zillow.net/browse/AIP-1608) as this would need to be changed in [aip-kfp-sdk](https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/aip-workflow/aip-kfp-sdk/blob/fc94616f6e25a942e75e9532a042eef4f02f99b8/aip_kfp_sdk/utils/aws.py#L15) as well. Added more details with links to the open MR in the description of [AIP-1608](https://zbrt.atl.zillow.net/browse/AIP-1608)

Reply:
Looks like a long standing issue in boto3 to be honest with no ETA for a fix: https://github.com/boto/boto3/issues/443

I'd say we just close out AIP-1608 as it's unlikely we'll improve this until AWS fixes boto3.

Reply:
Got you! Mentioned this there and closed it